### PR TITLE
[CARBONDATA-1073] Support INPUT_FILES

### PR DESCRIPTION
**Pull request by Sephiroth-Lin**
Currently we support INPUT_SEGMENT_NUMBERS to reading from specified segments, but will reading all data for one segment, in some scenario we need to reading from specified files, so add INPUT_FILES to support it.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2115/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2116/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2117/

**Comment by Sephiroth-Lin** [comment]
retest this please

**Comment by Sephiroth-Lin** [comment]
retest this please

**Comment by Sephiroth-Lin** [comment]
retest this please

**Comment by Sephiroth-Lin** [comment]
@QiangCai please help review

File: hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonInputFormat.java [hunk 1]
```diff
@@ -542,7 +554,7 @@ private boolean isValidBlockBasedOnUpdateDetails(
               updateStatusManager.getInvalidTimestampRange(segmentId), updateStatusManager,
               segmentId, validTaskKeys);
       if (!tableBlockInfoList.isEmpty()) {
-        // getFileStatusOfSegments(job, new int[]{ segmentId }, fileStatusList);
+        // getFileStatus(job, new String[]{ segmentId }, new String[] { fileName }, fileStatusList);
```
Comment (jackylk) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonInputFormat.java]: if it is not used, remove it

**Comment by jackylk** [comment]
Please add description in this PR

**Comment by Sephiroth-Lin** [comment]
@jackylk done

**Comment by Sephiroth-Lin** [comment]
retest this please

**Comment by Sephiroth-Lin** [comment]
retest this please

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**